src/Avalonia.Controls/RadioButton.cs

```diff
@@ -2,19 +2,144 @@
 // Licensed under the MIT license. See licence.md file in the project root for full license information.
 
 using System;
+using System.Collections.Generic;
 using System.Linq;
 using Avalonia.Controls.Primitives;
+using Avalonia.Rendering;
 using Avalonia.VisualTree;
 
 namespace Avalonia.Controls
 {
     public class RadioButton : ToggleButton
     {
+        private class RadioButtonGroupManager
+        {
+            public static readonly RadioButtonGroupManager Default = new RadioButtonGroupManager();
+            static readonly List<(WeakReference<IRenderRoot> Root, RadioButtonGroupManager Manager)> s_registeredVisualRoots
```

Comment:
Id suggest using the ConditionalWeakTable type here. It will automatically handle releasing the references for us. It will also allow us to simplify the GetOrCreateForRoot method significantly.

Reply:
Thanks for reminding me! This is the type I was looking for, but could not remember the name.